kb/2023-08-21/compatible_with_ip_pool_new_feature.md

```diff
@@ -0,0 +1,49 @@
+slug: upgrading_guest_clusters_with_harvester_ip_pool_compatibility
+authors:
+- name: Canwu Yao
+  title: Software Engineer
+  url: https://github.com/yaocw2020
+  image_url: https://avatars.githubusercontent.com/u/7421463?s=400&v=4
+tags: [harvester, load balancer, cloud provider, ip pool, upgrade] 
+hide_table_of_contents: false
+---
+
+As the **Harvester v1.2.0** releases, a new Harvester cloud provider version **0.2.2** is integrated into the RKE2 **v1.27.3+rke2r1**, **v1.26.6+rke2r1**, **v1.25.11+rke2r1** and **v1.24.15+rke2r1**.
+
+With Harvester v1.2.0, the new Harvester cloud provider provides more powerful load balancing for Kubernetes services. Harvester IP Pool, a built-in IP address management (IPAM) solution for the Harvester load balancer, allows users to provide an IP pool confined to the specific guest cluster by specifying the guest cluster name. For example, the following IP pool is only available for the guest cluster `cluster2`:
+
+![image](ippoolforcluster2.png)
+
+However, the feature is not compatible with the existing guest clusters even after upgrading because they didn't pass the cluster name to the Harvester cloud provider(refer to [issue 4232](https://github.com/harvester/harvester/issues/4232)). As a workaround, users can manually upgrade the Harvester cloud provider using Helm and pass the cluster name after upgrading, but this will change the IPs of the load balancer. This article will provide a workaround to allow users to use the new IP pool feature while keep the load balancer IPs unchanged.
+
+## Prerequisites
+- Download Harvester kubeconfig file from the Harvester UI, not from the Rancher UI if the Harvester has been imported to Rancher. Refer to [Access Harvester Cluster](https://docs.harvesterhci.io/v1.1/faq#how-can-i-access-the-kubeconfig-file-of-the-harvester-cluster) for more details.
+  
+- Download guest cluster kubeconfig file. Refer to [Accessing Clusters with kubectl from Your Workstation](https://ranchermanager.docs.rancher.com/how-to-guides/new-user-guides/manage-clusters/access-clusters/use-kubectl-and-kubeconfig#accessing-clusters-with-kubectl-from-your-workstation) to download the kubeconfig file.
+
+## Steps to Keep Load Balancer IP
+1. Execute the script before ugrading.
+   ```
+   curl -sfL https://raw.githubusercontent.com/harvester/harvesterhci.io/main/kb/2023-08-21/keepip.sh | sh -s before_upgrade <Harvester-kubeconfig-path> <guest-cluster-kubeconfig-path> <guest-cluster-name> <guest-cluster-nodes-namespace>
+   ```
+
+  The guest cluster nodes namespace is the namespace where the VMs composed of the guest cluster are located.
+  
+
+2. Add network selector for the pool. 
+  For example, the following cluster is under the VM network `default/mgmt-untagged`. The network selector should be `default/mgmt-untagged`.
+
+  ![image](network.png)
+
+  ![image](network-selector.png)
+
+3. Upgrade the RKE2 cluster.
+
+4. Execute the script after upgrading.
```

Comment:
what's the desired output, and how to check if the upgrade is working as expected?

Reply:
I add the `set -e` to terminate the script if an error occurs.

Reply:
I mean we should add an example output here so people can use it as a reference, also need to specify what has been changed after the script is executed correctly and how could users check it.